docs/dcp2_system_design.rst

```diff
@@ -0,0 +1,1465 @@
+
+   a) omit any object that hasn't change
+
+   b) (stricter) require that subgraphs are fully defined
+
+   c) like b but allow omission of data files
+
+   d) like c but also allow omission of descriptors
+
+WIP: Updates to data files
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Updates to data files are out of scope for MVP but shouldn't be too hard
+to implement. [TODO: revise]
+
+WIP: Deletion of data files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+WIP: Purging of data files
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[When donor consent is withdrawn or it was never given.]
+
+WIP: Renames of data files
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+WIP: Deletion of metadata entities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+WIP: Updates to sub-graphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[Doesn't the deletion of entities and data files imply a subgraph
+update? ]Updates to the metadata graph structure are out of scope for
+now [Maybe not] but shouldn't be too hard to implement via updates to
+``links/{links_id}_{version}.json``. In fact, they should work out of the
+box.
+
+WIP: Deletion of subgraphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TODO
+
+Project-level matrices
+======================
+
+Contributor-generated matrices (CGMs)
+-------------------------------------
+
+For DCP/2 MVP, an interim solution is used to store matrices provided by
+data contributors (as opposed to the matrices generated by the DCP/2)
+and describe them with just enough metadata so they can be indexed by
+Azul and exposed for direct download in the Data Browser. The artifacts
+(data, metadata, user interface elements) involved in this interim
+solution will be replaced with those produced by a longer-lived and more
+informationally rich solution, one that is mostly hinged upon devising
+an adequate metadata schema by which to describe these matrices.
+Reproducibility of analysis that uses the interim artefacts will be
+provided via TDR snapshots. The MVP snapshot will permanently expose the
+interim artifacts, while a later release snapshot will expose the
+permanent replacements.
+
+For DCP/2 MVP, no post-processing is performed on contributor-generated
+matrices. Users will be able to download them in exactly the same file
+format as they were provided by the contributor. For example, if the
+contributor provided a ZIP archive of multiple matrix files, the user
+will need to download that archive and extract it locally.
+
+The interim solution stores the contributor-generated matrices as data
+files in the TDR and describes them with a `file
+descriptor <Generic file metadata_>`_, a ``supplementary_file`` entity
+and a ``supplementary_link`` entry in a ``links.json``.
+There will be one subgraph (and therefore one ``links.json``
+document) per HCA project. If the contributor provided multiple matrices
+stratified by, say, species and organ, the per-project subgraph will
+contain a ``supplementary_file`` and a
+``supplementary_link`` for each such matrix file.
+
+.. image:: media/project-cgm.png
```

Comment:
(nitpick) can the image be centered?

Reply:
Markup added to center the image didn't seem to carry over to the displayed document. Not a major issue however.
![Screen Shot 2021-03-10 at 11 02 35 AM](https://user-images.githubusercontent.com/46407464/110682742-23278b80-8190-11eb-8d38-297f09093fcd.png)